Realm/Realm/Handles/ThreadSafeReferenceHandle.cs

```diff
@@ -27,16 +27,30 @@ private static class NativeMethods
         {
             [DllImport(InteropConfig.DLL_NAME, EntryPoint = "thread_safe_reference_destroy", CallingConvention = CallingConvention.Cdecl)]
             public static extern void destroy(IntPtr handle);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_thread_safe_reference_destroy", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void destroy_realm_reference(IntPtr handle);
         }
 
+        private bool _isRealmReference;
+
         [Preserve]
-        public ThreadSafeReferenceHandle(IntPtr handle) : base(null, handle)
+        public ThreadSafeReferenceHandle(IntPtr handle, bool isRealmReference = false) : base(null, handle)
```

Comment:
This is quite ugly, but couldn't find a better way to do it.

Reply:
It's going to go away with Core 6 soon, so we'll just live with it for a while.